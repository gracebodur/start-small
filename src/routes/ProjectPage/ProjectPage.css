.ProjectPage__image {
    height: 20vw;
    min-height: 150px;
    background-size: cover;
    background-position: 50% 50%;
    border-radius: 10px 10px 0 0;
  }
  
  .ProjectPage__star-list {
    list-style-type: none;
    padding: 0;
  }
  
  .ProjectPage__content {
    margin-bottom: 30px;
  }
  
  .ProjectPage__review {
    position: relative;
  }
  
  .ProjectPage__review::before {
    border-color: transparent #f5f5f5;
    border-style: solid;
    border-width: 0px 0px 10px 10px;
    content: '';
    display: block;
    height: 0px;
    position: absolute;
    width: 0;
    bottom: 14px;
    right: 7px;
  }
  
  .ProjectPage__review-icon {
    font-size: 12pt;
    margin-right: 10px;
    vertical-align: super !important;
  }
  
  .ProjectPage__review-text {
    background: #f5f5f5;
    padding: 8px;
    margin-bottom: 4px;
    border-radius: 4px;
  }
  
  .ProjectPage__review-user {
    color: #666;
    margin: 8px 0 0;
    font-size: 10pt;
    font-weight: bold;
    text-align: right;
  }
  
  .ProjectPage__review-user .ProjectStarRating {
    display: inline;
  }
  
  .Fundurl {
    position: absolute;
    right: 35vh;
  }

  .ProjectPage__review-list {
    color: red;
    margin-top: 25vh;
  }
