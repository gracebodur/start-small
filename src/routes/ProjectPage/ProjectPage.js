import React, { Component } from 'react'
import ProjectContext from '../../contexts/ProjectContext'
import ProjectsApiService from '../../services/projects-api-service'
import { Section } from '../../components/Utils/Utils'
import { ProjectStarRating } from '../../components/ProjectStarRating/ProjectStarRating'
import ShareButton from '../../components/ShareButton/ShareButton'
import ProjectReviewForm from '../../components/ProjectReviewForm/ProjectReviewForm'
import donorsLogo from '../../img/donors-logo.png'
import { Scrollbars } from 'rc-scrollbars'
import './ProjectPage.css'

class ProjectPage extends Component {
	static defaultProps = {
		match: { params: {} },
	}

	static contextType = ProjectContext

	componentDidMount() {
		const { project_id } = this.props.match.params
		this.context.clearError()
		ProjectsApiService.getById(project_id).then(this.context.setProject).catch(this.context.setError)
		ProjectsApiService.getProjectReviews(project_id)
			.then(this.context.setReviews)
			.catch(this.context.setError)
	}

	componentWillUnmount() {
		this.context.clearProject()
	}

	renderProject() {
		const { project, reviews } = this.context
		return (
			<Section>
				<div className='container py-3 project-wrapper'>
					<div className='card'>
						<div className='row '>
							<div className='col-md-4'>
								<img src={project.imageurl} className='fadeIn card-img' alt='project description' />
							</div>
							<div className='col-md-8 px-3'>
								<div className='card-block px-3'>
									<h1 className='card-title'>{project.schoolname}</h1>
									<h3 className='card-text'>"{project.fulfillmenttrailer}"</h3>
									<h3 className='card-text'>-{project.teachername}</h3>
									<div className='m-3'></div>
									<a href={project.fundurl} target='_blank' rel='noopener noreferrer'>
										<img
											className='donors__logo mr-3'
											alt='donation website'
											src={donorsLogo}
											style={{ width: '15%' }}
										/>
									</a>
									<ShareButton />
								</div>
							</div>
						</div>
					</div>
				</div>
				<div className='container'>
					<div className='row'>
						<div className='col-sm-6'>
							<ProjectReviews reviews={reviews} />
						</div>
						<div className='col-sm-6'>
							<ProjectReviewForm />
						</div>
					</div>
				</div>
			</Section>
		)
	}

	render() {
		const { error, project } = this.context
		let content
		if (error) {
			content =
				error.error === `Project doesn't exist` ? (
					<p className='red text-center'>Project not found</p>
				) : (
					<p className='red text-center'>There was an error</p>
				)
		} else if (!project.project_id) {
			content = <div className='loading' />
		} else {
			content = this.renderProject()
		}
		return <Section className='ProjectPage'>{content}</Section>
	}
}

function ProjectReviews({ reviews = [] }) {
	return (
		<Section>
			<div className='text-center'>
				<h1 className='h1 review-title'>Reviews:</h1>
			</div>
					<Scrollbars style={{ height: 400 }}>
						<ul className='list-group'>
							{reviews.map((review) => (
								<li className='list-group-item text-center' key={review.review_id}>
									<h3>"{review.text}"</h3>
									<h5>- {review.user.full_name}</h5>
									<ProjectStarRating rating={review.rating} />
								</li>
							))}
						</ul>
					</Scrollbars>
		</Section>
	)
}

export default ProjectPage
