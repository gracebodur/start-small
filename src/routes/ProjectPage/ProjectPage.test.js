import React from "react";
import toJson from "enzyme-to-json";
import ProjectPage from "./ProjectPage";

describe(`ProjectPage component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a ProjectPage by default", () => {
    const wrapper = <ProjectPage />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the ProjectPage given props", () => {
    const wrapper = <ProjectPage {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
