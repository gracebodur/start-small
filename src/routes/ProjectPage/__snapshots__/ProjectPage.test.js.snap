// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPage component renders a ProjectPage by default 1`] = `undefined`;

exports[`ProjectPage component renders the ProjectPage given props 1`] = `undefined`;
